react/SearchWrapper.tsx

```diff
@@ -0,0 +1,186 @@
+import useDataPixel from './hooks/useDataPixel'
+import { usePageView } from './components/PageViewPixel'
+import { getDepartmentMetadata, getCategoryMetadata, getSearchMetadata } from './modules/searchMetadata'
+import { SearchQuery } from './modules/searchTypes'
+import { PixelEvent } from './typings/event'
+
+const APP_LOCATOR = 'vtex.store'
+
+interface SearchRouteParams {
+  term?: string
+  department?: string
+  category?: string
+}
+
+const pageCategory = (products: unknown[], params: SearchRouteParams) => {
+  if (!products || products.length === 0) {
+    return 'EmptySearch'
+  }
+  const { category, term } = params
+  return term ? 'InternalSiteSearch' : category ? 'Category' : 'Department'
+}
+
+type PageEventName = 'internalSiteSearchView' | 'categoryView' | 'departmentView' | 'emptySearchView'
+
+const mapEvent = {
+  InternalSiteSearch: 'internalSiteSearchView',
+  Category: 'categoryView',
+  Department: 'departmentView',
+  EmptySearch: 'emptySearchView',
+}
+const fallbackView = 'otherView'
+
+const getPageEventName = (products: unknown[], params: SearchRouteParams) : PageEventName => {
+  if (!products || products.length === 0) {
+    return fallbackView as PageEventName
+  }
+  const category = pageCategory(products, params)
+
+  return (mapEvent[category] || fallbackView) as PageEventName
+}
+
+const getTitleTag = (titleTag: string, storeTitle: string, term?: string) => {
+  return titleTag
+    ? `${titleTag} - ${storeTitle}`
+    : term
+    ? `${capitalize(decodeURI(term))} - ${storeTitle}`
+    : `${storeTitle}`
+}
+
+const getSearchIdentifier = (searchQuery: SearchQuery) => {
+  const variables = searchQuery.variables
+  if (!variables) {
+    return
+  }
+  const { query, map } = variables
+  return query + map
+}
+
+interface SearchPageInfoEvent {
+  department: DepartmentMetadata | null | undefined
+  category: CategoryMetadata | null | undefined
+  search: SearchMetadata | null | undefined
+}
+
+interface DepartmentMetadata {
+  id: string
+  name: string
+}
+
+interface CategoryMetadata {
+  id: string
+  name: string
+}
+
+interface SearchMetadata {
+  term: string
+  category: string | null
+  results: number
+}
+
+interface SearchWrapperProps {
+  children: ReactElement,
+  params: SearchRouteParams,
+  searchQuery: SearchQuery
+}
+
+const SearchWrapper: FC<SearchWrapperProps> = props => {
+  const {
+    params,
+    searchQuery,
+    searchQuery: {
+      loading = true,
+      data: { searchMetadata: { titleTag = '', metaTagDescription = '' } = {} } = {},
+    } = {},
+    children,
+  } = props
+  const { account, getSettings } = useRuntime()
+  const settings = getSettings(APP_LOCATOR) || {}
+  const { titleTag: defaultStoreTitle, storeName } = settings
+  const title = getTitleTag(
+    titleTag,
+    storeName || defaultStoreTitle,
+    params.term
+  )
+
+  const pixelEvents = useMemo(() => {
+    if (
+      !searchQuery ||
+      typeof document === 'undefined' ||
```

Comment:
You could use `canUseDOM` from `runtime` here? I'm not sure how that is implemented. If it does exactly the same thing, might just be a matter of preference and you can ignore this hahahah

Reply:
True. I didn't change this bit of code, so I didn't reflect on that.